pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -0,0 +1,206 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/aws/utils"
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+	"github.com/patrickmn/go-cache"
+	"go.uber.org/zap"
+)
+
+type InstanceTypeProvider struct {
+	ec2               ec2iface.EC2API
+	vpc               *VPCProvider
+	instanceTypeCache *cache.Cache
+}
+
+func NewInstanceTypeProvider(ec2 ec2iface.EC2API, vpcProvider *VPCProvider) *InstanceTypeProvider {
```

Comment:
I'd call arg `ec2api` to avoid conflict with import on line 22.

Reply:
👍  done!